components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -849,6 +851,13 @@ public String createApplication(ApplicationModel applicationModel, String templa
     public void patchApplication(String applicationId, ApplicationPatchModel applicationPatchModel) {
 
         ServiceProvider appToUpdate = cloneApplication(applicationId);
+        AssociatedRolesConfig associatedRoles = applicationPatchModel.getAssociatedRoles();
+        if (associatedRoles != null) {
+            List<Role> listRole = associatedRoles.getRoles();
+            if (listRole.isEmpty()) {
+                throw buildClientError(ErrorMessage.INVALID_ROLE_ASSOCIATION_FOR_ORGANIZATION_AUDIENCE);
```

Comment:
since we are setting the organization audience as the default audience of the app, here we expect the all the organization roles? not clear here

Reply:
For both application roles and organization roles, we have restricted the app association using the rest application update API.